### PR TITLE
Fix support for *-empty-line-before in HTML

### LINKED ISSUE #2853: Fix false positives for CSS in HTML in *-empty-line-before
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

`*-empty-line-before` not support for HTML

> Which rule, if any, is this issue related to?

`rule-empty-line-before`, `at-rule-empty-line-before`

> What CSS is needed to reproduce this issue?

It is CSS in HTML.

```html
<style>
h1 {
  font-weight: normal;
}
</style>
```

> What stylelint configuration is needed to reproduce this issue?

```js
module.exports = {
  extends: 'stylelint-config-standard',
  // add your custom rules here
  rules: {
  }
};

```

> Which version of stylelint are you using?

`8.1.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

```shell
stylelint "**/*.{vue,html,css,sss,less,scss,sass}" --custom-syntax postcss-html
```

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

no.

> What did you expect to happen?

"No warnings to be flagged."

> What actually happened (e.g. what warnings or errors you are getting)?

"The following errors were flagged:"
https://circleci.com/gh/gucong3000/webpack/16#artifacts/containers/0

```shell
src/App.vue
 15:1  ✖  Expected empty line before rule   rule-empty-line-before

src/components/Hello.vue
 36:1  ✖  Expected empty line before rule   rule-empty-line-before
```

### PR DESCRIPTION
**Pull request by gucong3000**
> Which issue, if any, is this issue related to?

#2853

> Is there anything in the PR that needs further explanation?

"No, it's self explanatory."

### CODE REVIEW

File: lib/rules/at-rule-empty-line-before/index.js [hunk 1]
```diff
@@ -59,7 +59,7 @@ const rule = function(expectation, options, context) {
       const isNested = atRule.parent !== root;
 
       // Ignore the first node
-      if (atRule === root.first) {
+      if (atRule === root.first || atRule === atRule.root().first) {
```
Comment (evilebottnawi) [inline lib/rules/at-rule-empty-line-before/index.js]: Please add comment around `atRule === atRule.root().first` and good work!

Comment (gucong3000) [reply lib/rules/at-rule-empty-line-before/index.js]: Done.

File: lib/rules/rule-empty-line-before/index.js [hunk 2]
```diff
@@ -51,8 +51,12 @@ const rule = function(expectation, options, context) {
         return;
       }
 
-      // Ignore the first node
-      if (rule === root.first) {
+      if (
```
Comment (jeddy3) [inline lib/rules/rule-empty-line-before/index.js]: Can we please move this into a `util`, e.g. `isFirstNodeOfRoot(node)` as I suspect other rules will need too.

File: lib/rules/rule-empty-line-before/index.js [hunk 3]
```diff
@@ -52,7 +53,7 @@ const rule = function(expectation, options, context) {
       }
 
       // Ignore the first node
-      if (rule === root.first) {
+      if (isFirstNodeOfRoot(rule, root)) {
```
Comment (jeddy3) [inline lib/rules/rule-empty-line-before/index.js]: Did you mean to pass `root` in as a second parameter?

Comment (gucong3000) [reply lib/rules/rule-empty-line-before/index.js]: It was a mistake.

**Comment by jeddy3** [comment]
> -   Fixed: `*-empty-line-before` false positives for CSS in HTML ([#2854](https://github.com/stylelint/stylelint/issues/2854)).

**Pull request merged**